### PR TITLE
Core UI tests

### PR DESCRIPTION
**Pull request by cobbal**
Pulled CanvasUITests out into a separate framework under Core for inclusion in both the Teacher and Student UI test suite.

Added some machinery for a test class to be "abstract" meaning that
while it may define tests, they will only be run for the subclasses.

One test on Teacher currently causes a RN crash.

### CODE REVIEW

File: Core/CoreUITests/Assignments/AssignmentsTests.swift [hunk 1]
```diff
@@ -17,9 +17,11 @@
 //
 
 import XCTest
-import TestsFoundation
+@testable import TestsFoundation
+
+class AssignmentsTests: CoreUITests {
+    override var abstractTestClass: CoreUITests.Type { return AssignmentsTests.self }
```
Comment (vanwagonet) [inline Core/CoreUITests/Assignments/AssignmentsTests.swift]: This test is specific to the Student app, and should probably be moved to StudentUITests.

File: Core/CoreUITests/Calendar/CalendarTests.swift [hunk 2]
```diff
@@ -29,7 +29,9 @@ enum Calendar {
     }
 }
 
-class CalendarTests: CanvasUITests {
+class CalendarTests: CoreUITests {
+    override var abstractTestClass: CoreUITests.Type { return CalendarTests.self }
```
Comment (vanwagonet) [inline Core/CoreUITests/Calendar/CalendarTests.swift]: StudentUITests

File: Core/CoreUITests/Course Files/CourseFileTests.swift [hunk 3]
```diff
@@ -25,7 +25,9 @@ enum FilesList {
     }
 }
 
-class CourseFileTests: CanvasUITests {
+class CourseFileTests: CoreUITests {
+    override var abstractTestClass: CoreUITests.Type { return CourseFileTests.self }
```
Comment (vanwagonet) [inline Core/CoreUITests/Course Files/CourseFileTests.swift]: this is shared

File: Core/CoreUITests/Modules/ModulesTests.swift [hunk 4]
```diff
@@ -19,7 +19,8 @@
 import XCTest
 import TestsFoundation
 
-class ModulesTests: CanvasUITests {
+class ModulesTests: CoreUITests {
```
Comment (vanwagonet) [inline Core/CoreUITests/Modules/ModulesTests.swift]: StudentUITests

File: Core/CoreUITests/Notifications/NotificationsListTests.swift [hunk 5]
```diff
@@ -20,7 +20,8 @@ import XCTest
 import TestsFoundation
 @testable import Core
 
-class NotificationsListTests: CanvasUITests {
+class NotificationsListTests: CoreUITests {
```
Comment (vanwagonet) [inline Core/CoreUITests/Notifications/NotificationsListTests.swift]: StudentUITests

File: Core/CoreUITests/People/PeopleTests.swift [hunk 6]
```diff
@@ -31,7 +31,9 @@ enum PersonContextCard {
     }
 }
 
-class PeopleTests: CanvasUITests {
+class PeopleTests: CoreUITests {
```
Comment (vanwagonet) [inline Core/CoreUITests/People/PeopleTests.swift]: StudentUITests

File: Core/CoreUITests/Quizzes/QuizTests.swift [hunk 7]
```diff
@@ -19,7 +19,9 @@
 import XCTest
 import TestsFoundation
 
-class QuizTests: CanvasUITests {
+class QuizTests: CoreUITests {
```
Comment (vanwagonet) [inline Core/CoreUITests/Quizzes/QuizTests.swift]: StudentUITests

File: Core/CoreUITests/CoreUITestCase.swift [hunk 8]
```diff
@@ -20,7 +20,7 @@ import Core
 import XCTest
 import TestsFoundation
 
-class CoreUITests: UITestCase {
+class CoreUITestCase: UITestCase {
```
Comment (vanwagonet) [inline Core/CoreUITests/CoreUITestCase.swift]: I like this name better. 👍

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a release note. If no release note is wanted, use `none`. Example: `release note: Fixed a bug that prevented users from enjoying the app.`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add which apps this change affects. Example: `affects: Teacher, Student` or `affects: none`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a reference to a jira ticket. For example: `refs: MBL-10023`
  </td>
    </tr>
  </tbody>
</table>



Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 94.97% | 94.86% | 0.11%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**